File: SEP3_Database/src/main/java/com/example/SEP3_Database/model/domain/list/postList/PostList.java
package com.example.SEP3_Database.model.domain.list.postList;

import com.example.SEP3_Database.model.domain.unit.post.Post;

import java.util.ArrayList;

public class PostList {
    private ArrayList<Post> Posts;

    public PostList()
    {
        Posts = new ArrayList<>();
    }

    public PostList(ArrayList<Post> postList)
    {
        Posts = new ArrayList<>();
        for (int x=0;x<postList.size();x++)
        {
            Posts.add(postList.get(x).copy());
        }
    }

    public boolean hasPostId(String id)
    {
        for (int x=0;x<Posts.size();x++)
        {
            if (Posts.get(x).getPostId()==id)
            {
                return true;
            }
        }
        return false;
    }

    public String addPost(Post post)
    {
        if (post!=null)
        {
            if (!hasPostId(post.getPostId()))
            {
                Posts.add(post);
            }
            else
            {
                return "Not a new post.";
            }
        }
        return "Input null.";
    }

    public Post getPostByPostId(String id)
    {
        for (int x=0;x<Posts.size();x++)
        {
            if (Posts.get(x).getPostId()==id)
            {
                return Posts.get(x);
            }
        }
        return null;
    }

    public Post getPostByIndex(int index)
    {
        if (index>=0&&index<Posts.size())
        {
            return Posts.get(index);
        }
        return null;
    }

    public PostList getPostBySenderId(String id)
    {
        var postList = new PostList();
        for (int x=0;x<Posts.size();x++)
        {
            if (Posts.get(x).getPostId()==id)
            {
                postList.addPost(Posts.get(x));
            }
        }
        return postList;
    }

    public String updatePost(Post post)
    {
        var target = getPostByPostId(post.getPostId());
        if (target!=null)
        {
            target = new Post(post);
            return null;
        }
        return "Not find post.";
    }

    public void removePostByPostId(String id)
    {
        for (int x=0;x<Posts.size();x++)
        {
            if (Posts.get(x).getPostId()==id)
            {
                Posts.remove(x);
                break;
            }
        }
    }

    public PostList copy()
    {
        return new PostList(Posts);
    }
}

File: SEP3_PostSystem/Model/List/PostList/PostList.cs
using System.Collections.Generic;
using System.Linq;
using SEP3_PostSystem.Model.Unit.Post;

namespace SEP3_PostSystem.Model.List.PostList
{
    public class PostList
    {
        public List<Post> Posts { get; set; }
        
        public PostList()
        {
            Posts = new List<Post>();
        }

        public PostList(List<Post> postList)
        {
            Posts = new List<Post>();
            foreach (var post in postList)
            {
                Posts.Add(post.Copy());
            }
        }
        
        public bool HasPostId(string id)
        {
            return Posts.Any(post => post.PostId == id);
        }

        public string AddPost(Post post)
        {
            if (post!=null)
            {
                if (!HasPostId(post.PostId))
                {
                    Posts.Add(post);
                    return null;
                }
                else
                {
                    return "Not a new post.";
                }
            }
            return "Input null.";
        }

        public Post GetPostByPostId(string id)
        {
            return Posts.FirstOrDefault(post => post.PostId == id);
        }
        
        public Post GetPostByIndex(int index)
        {
            if (index>=0&&index<Posts.Count)
            {
                return Posts[index];
            }
            return null;
        }
        
        public PostList GetPostBySenderId(string id)
        {
            var postList = new PostList();
            foreach (var post in Posts.Where(post => post.SenderId==id))
            {
                postList.Posts.Add(post);
            }

            return postList;
        }

        public string UpdatePost(Post newPost)
        {
            var target = GetPostByPostId(newPost.PostId);
            if (target!=null)
            {
                target = new Post(newPost);
                return null;
            }
            return "Not find post.";
        }

        public void RemovePostByPostId(string id)
        {
            foreach (var post in Posts.Where(post => post.PostId==id))
            {
                Posts.Remove(post);
                break;
            }
        }

        public PostList Copy()
        {
            return new PostList(Posts);
        }
    }
}